src/Umbraco.Infrastructure/Packaging/PackageInstallation.cs

```diff
@@ -44,20 +44,29 @@ public InstallationSummary InstallPackageData(CompiledPackage compiledPackage, i
             //make sure the definition is up to date with everything
             foreach (var x in installationSummary.DataTypesInstalled)
                 packageDefinition.DataTypes.Add(x.Id.ToInvariantString());
+
             foreach (var x in installationSummary.LanguagesInstalled)
                 packageDefinition.Languages.Add(x.Id.ToInvariantString());
+
             foreach (var x in installationSummary.DictionaryItemsInstalled)
                 packageDefinition.DictionaryItems.Add(x.Id.ToInvariantString());
+
             foreach (var x in installationSummary.MacrosInstalled)
                 packageDefinition.Macros.Add(x.Id.ToInvariantString());
+
             foreach (var x in installationSummary.TemplatesInstalled)
                 packageDefinition.Templates.Add(x.Id.ToInvariantString());
+
             foreach (var x in installationSummary.DocumentTypesInstalled)
                 packageDefinition.DocumentTypes.Add(x.Id.ToInvariantString());
-            foreach (var x in installationSummary.StylesheetsInstalled)
-                packageDefinition.Stylesheets.Add(x.Id.ToInvariantString());
-            var contentInstalled = installationSummary.ContentInstalled.ToList();
-            packageDefinition.ContentNodeId = contentInstalled.Count > 0 ? contentInstalled[0].Id.ToInvariantString() : null;
+
+            foreach (var x in installationSummary.MediaTypesInstalled)
+                packageDefinition.MediaTypes.Add(x.Id.ToInvariantString());
+
+            packageDefinition.ContentNodeId = installationSummary.ContentInstalled.FirstOrDefault()?.Id.ToInvariantString();
+            
+            foreach (var x in installationSummary.MediaInstalled)
+                packageDefinition.MediaUdis.Add(x.GetUdi());
```

Comment:
I'm not sure where the updated package definition is used, but I've added media and removed the stylesheets as they aren't tracked by ID.

So the following types aren't currently tracked, because that would require serializing them to XML (that includes the path and file contents):
- `MacroPartialViewsInstalled`
- `StylesheetsInstalled`
- `ScriptsInstalled`
- `PartialViewsInstalled`

Reply:
The package definition is now also updated with the imported stylesheets, scripts and partial views paths. Macro partial views are part of the macro, so isn't defined as separate entity in the package definition, which is fine I guess.